### PR TITLE
[[Bug 15774]] Add filter to script editor handler list.

### PR DESCRIPTION
**Pull request by peter-b**
This is really helpful for navigating around huge scripts!

This PR has three changes:
- Moving some existing SE code into a script-only stack
- Adding the filter field object
- Adding its behaviour and hooking up the filtering.

### CODE REVIEW

**Comment by peter-b** [comment]
I need to double check with @techstrategies that he's signed the CLA.

**Comment by peter-b** [comment]
I've confirmed that @techstrategies has signed the CLA.

File: Toolset/palettes/script editor/behaviors/revsehandlerfilterbehavior.livecodescript [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+private command selectAll
+   select char 1 to -1 of me
+end selectAll
+
+-- This does the main work of fixing up the formatting.  It also
+-- sends the registered callback message if necessary.
+on update
+   if the cFiltering of me then
+      set the textstyle of me to "normal"
```
Comment (livecodeali) [inline Toolset/palettes/script editor/behaviors/revsehandlerfilterbehavior.livecodescript]: I don't think "normal" is a valid textStyle (although I believe invalid text styles default to empty anyway). Either "plain" or empty gives unstyled text.

File: Toolset/palettes/script editor/behaviors/revseleftbarbehavior.livecodescript [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+   if tHandler is not empty then
+      set the cSelectedHandler of tObject to tHandler
+   end if
+   
+   send "updateSelectedHandler" to tObject
+end updateSelectedHandler
+
+# Description
+#   Callback for the handler list object. This is called when the stack is closed
+#   which either means the user has choosen a handler, or dismissed the stack.
```
Comment (livecodeali) [inline Toolset/palettes/script editor/behaviors/revseleftbarbehavior.livecodescript]: typo here (choosen)

**Pull request merged**